motoman_mh_support/urdf/mh180_120_macro.xacro

```diff
@@ -140,28 +140,28 @@
       <origin xyz="0 0 1.150" rpy="0 0 0"/>
       <parent link="${prefix}link_l"/>
       <child link="${prefix}link_u"/>
-      <axis xyz="0 1 0"/>
-      <limit lower="-${radians(142)}" upper="${radians(90)}" effort="0" velocity="${radians(125)}"/>
+      <axis xyz="0 -1 0"/>
+      <limit lower="-${radians(85)}" upper="${radians(90)}" effort="0" velocity="${radians(125)}"/>
```

Comment:
This is quite a change for a joint limit. @Jmeyer1292: was this a mistake by @marip8 when he first modelled the robot, or is this based on joint limits of your particular robot?

Reply:
This is the limit of a fresh off the pallette robot but it could have been set up differently by Yaskawa. The spec sheet says 142.

There is a cable tray on the back of joint 3 that would interfere with the upper arm if the robot could been it's elbow past 90. I can take pictures if you want.

Reply:
I believe you, but it's just that we (afaik) never deviate from the spec-sheet.

Specific dress-outs could need different limits, but that should then be mentioned in the `package.xml`. Otherwise it's hard for users to figure out which specific set of limits was used for the xacro.

Reply:
I've noticed that the joint limits of joint 3 on a couple of Motoman robots that we have been working with recently (MH180-120, ES200) are specified relative to a fixed vertical plane but the actual/physical joint limit of the robot changes with respect to joint 2. I initially made the joint limit according to the spec sheet, but this doesn't really make sense for the URDF definition of joint limits which are fixed and relative to the "upstream" link.

Reply:
This almost sounds like there is some sort of coupling going on. @ted-miller: could you provide any info?

Reply:
I don't know why the spec sheet says -142 degrees for that arm.  I opened up that arm in MotoSimVRC and I agree with the -85 degrees.  The arm will collide on itself if it goes any further.

In regard to marip8's comment, the MH180-120 has a 'Special Mechanical Interference' limit. This is to prevent the extra-long arm from hitting itself in certain positions.  The attached image shows an example.

![special mechanical interferance](https://user-images.githubusercontent.com/4623811/36326799-65da73ca-132a-11e8-8a97-fee514e72a24.PNG)

Reply:
Currently, we are not doing anything in MotoROS to detect the Special Mech limits.  I expect that the controller will alarm if MotoROS attempts to cross the limits.

Reply:
@ted-miller wrote:
> I don't know why the spec sheet says -142 degrees for that arm. I opened up that arm in MotoSimVRC and I agree with the -85 degrees. The arm will collide on itself if it goes any further.

ok, that's good to know.

@marip8: perhaps we should make a not of the fact that the limits do not correspond to the spec sheet, but do correspond to MotoSim?